js/src/secureApi.js

```diff
@@ -25,12 +25,13 @@ export default class SecureApi extends Api {
     this._isConnecting = true;
     this._connectState = sysuiToken === 'initial' ? 1 : 0;
     this._needsToken = false;
-    this._nextToken = nextToken;
     this._dappsPort = 8080;
     this._dappsInterface = null;
     this._signerPort = 8180;
+    this._followConnectionTimeoutId = null;
 
-    console.log('SecureApi:constructor', sysuiToken);
+    // Try tokens from hash, then from localstorage
+    this._tokensToTry = [ nextToken, sysuiToken ].filter((t) => t && t.length);
```

Comment:
Needs to be localStorage first, otherwise we end up with a bunch of single-use tokens. (Parity needs to clean this up, so we want to re-use as much as possible.)